contracts/gateway/EIP20CoGateway.sol

```diff
@@ -912,23 +890,20 @@ contract EIP20CoGateway is GatewayBase {
         bytes32 intentHash = GatewayLib.hashRedeemIntent(
             _amount,
             _beneficiary,
-            msg.sender,
-            _nonce,
-            _gasPrice,
-            _gasLimit,
-            valueToken
+            address(this)
         );
 
-        // Get the messageHash.
-        messageHash_ = MessageBus.messageDigest(
-            REDEEM_TYPEHASH,
+        MessageBus.Message memory message = getMessage(
```

Comment:
Can we directly call `MessageBus.Message`? 
like its done in `confirmStakeIntent`

Reply:
We could, but the purpose of the method was that it sets the consumed gas to `0` for a new message. In the other places we have to do it directly, because otherwise we would get a "stack too deep". What's your preference?